Update from v4.3.x to v4.4 (#1870)
ezsystems/developer-documentation: docs/update_and_migration/from_4.3/update_from_4.3_new_commerce.md

### Update the application to v4.4

First, run:

=== "[[= product_name_content =]]"

    ``` bash
    composer require ibexa/content:[[= latest_tag_4_4 =]] --with-all-dependencies --no-scripts
    composer recipes:install ibexa/content --force -v
    ```

=== "[[= product_name_exp =]]"

    ``` bash
    composer require ibexa/experience:[[= latest_tag_4_4 =]] --with-all-dependencies --no-scripts
    composer recipes:install ibexa/experience --force -v
    ```

=== "[[= product_name_com =]]"

    ``` bash
    composer require ibexa/commerce:[[= latest_tag_4_4 =]] --with-all-dependencies --no-scripts
    composer recipes:install ibexa/commerce --force -v
    ```

The `recipes:install` command installs new YAML configuration files.
Review the old YAML files and move your custom configuration to the relevant new files

### Flysystem v2

Local adapters' `directory` key changed to `location`.
It is defined in `config/packages/oneup_flysystem.yaml`:

```yaml
oneup_flysystem:
    adapters:
        default_adapter:
            local:
                location: '%kernel.cache_dir%/flysystem'
```

If you haven't applied custom changes to that file,

> you can simply reset third party `oneup/flysystem-bundle` recipe by executing:

Reviewer: ```suggestion
you can reset the third-party `oneup/flysystem-bundle` recipe by executing:
```

```bash
composer recipe:install --force --reset -- oneup/flysystem-bundle
```

### Remove `ibexa/commerce-*` packages with dependencies

Remove the following bundles from `config/bundles.php`.

> Removal of third party bundles (`FOS\` to `JMS\`) is optional, you can leave them if they are used by your installation.

Reviewer: ```suggestion
You do not have to remove third-party bundles (`FOS\` to `JMS\`) if they are used by your installation.
```

    Ibexa\Bundle\Commerce\ShopUi\IbexaCommerceShopUiBundle
    # ...
    Ibexa\Bundle\Commerce\OneSky\IbexaCommerceOneSkyBundle
    Ibexa\Bundle\Commerce\EzStudio\IbexaCommerceEzStudioBundle
    Ibexa\Bundle\Commerce\Comparison\IbexaCommerceComparisonBundle
    Ibexa\Bundle\Commerce\QuickOrder\IbexaCommerceQuickOrderBundle
    Ibexa\Bundle\Commerce\TestTools\IbexaCommerceTestToolsBundle
    Ibexa\Bundle\Commerce\Voucher\IbexaCommerceVoucherBundle
    Ibexa\Bundle\Commerce\LocalOrderManagement\IbexaCommerceLocalOrderManagementBundle
    Ibexa\Bundle\Commerce\Newsletter\IbexaCommerceNewsletterBundle
    Ibexa\Bundle\Commerce\OrderHistory\IbexaCommerceOrderHistoryBundle
    Ibexa\Bundle\Commerce\ErpAdmin\IbexaCommerceErpAdminBundle
    Ibexa\Bundle\Commerce\ShopFrontend\IbexaCommerceShopFrontendBundle
    Ibexa\Bundle\Commerce\Basket\IbexaCommerceBasketBundle::class
    Ibexa\Bundle\Commerce\Rest\IbexaCommerceRestBundle::class
    Ibexa\Bundle\Commerce\AdminUi\IbexaCommerceAdminUiBundle::class
    Ibexa\Bundle\Commerce\PageBuilder\IbexaCommercePageBuilderBundle::class
    EWZ\Bundle\RecaptchaBundle\EWZRecaptchaBundle::class
    ```

Next, remove related extensions' configuration.

> Removal of related extensions' for third party bundles (for example `config/packages/fos_rest.yaml`) is optional, you can leave them if they are used by your installation.

Reviewer: ```suggestion
You do not have to remove third-party bundles (for example `config/packages/fos_rest.yaml`) if they are used by your installation.
```

    ```
    config/packages/commerce.yaml
    config/packages/commerce/autogenerated/.gitkeep
    config/packages/commerce/commerce.yaml
    config/packages/commerce/commerce_advanced.yaml
    config/packages/commerce/commerce_common.yaml
    config/packages/commerce/commerce_demo.yaml
    config/packages/commerce/commerce_parameters.yaml
    config/packages/nelmio_solarium.yaml
    ```

=== "[[= product_name_com =]]"

    ```
    config/packages/commerce.yaml
    config/packages/commerce/autogenerated/.gitkeep
    config/packages/commerce/commerce.yaml
    config/packages/commerce/commerce_advanced.yaml
    config/packages/commerce/commerce_common.yaml
    config/packages/commerce/commerce_demo.yaml
    config/packages/commerce/commerce_parameters.yaml
    config/packages/dev/ewz_recaptcha.yaml
    config/packages/dev/jms_serializer.yaml
    config/packages/ewz_recaptcha.yaml
    config/packages/ezcommerce/autogenerated/commerce_repository_parameters.yaml
    config/packages/fos_rest.yaml
    config/packages/google_recaptcha.yaml
    config/packages/jms_serializer.yaml
    config/packages/nelmio_solarium.yaml
    config/packages/prod/jms_serializer.yaml
    ```

> Finally, remove related routes from `config/routes/ibexa_commerce.yaml`.

Author: @adamwojs should we remove the whole file?

> Finally, remove related routes from `config/routes/ibexa_commerce.yaml`.

Reviewer 2: Whole file can be removed. It contains only a import.

> Finally, removed related routes from `config/routes/ibexa_commerce.yaml`.

Reviewer: ```suggestion
Finally, remove related routes from `config/routes/ibexa_commerce.yaml`.
```

### Update the database

> Next, update the database in you have a Commerce version of Ibexa DXP.

Reviewer: ```suggestion
Next, update the database if you are using Ibexa Commerce.
```

> Ibexa DXP Content and Ibexa DXP Experience do not require database update.

Reviewer: ```suggestion
Ibexa Content and Ibexa Experience do not require the database update.
```

[[% include 'snippets/update/db/db_backup_warning.md' %]]

Apply the following database update scripts:

=== "MySQL"

    ``` bash
    mysql -u <username> -p <password> <database_name> < vendor/ibexa/installer/upgrade/db/mysql/commerce/ibexa-4.3.latest-to-4.4.0.sql
    ```

=== "PostgreSQL"

    ``` bash

>     psql <database_name> < vendor/ibexa/installer/upgrade/db/postgresql/commerce/ibexa-4.3.latest-to-4.4.0.sqll

Reviewer 3: ```suggestion
    psql <database_name> < vendor/ibexa/installer/upgrade/db/postgresql/commerce/ibexa-4.3.latest-to-4.4.0.sql
```

    ```

#### Ibexa Open Source

If you have no access to Ibexa DXP's `ibexa/installer` package, database upgrade is not necessary.

## Ensure password safety

Following [Security advisory: IBEXA-SA-2022-009](https://developers.ibexa.co/security-advisories/ibexa-sa-2022-009-critical-vulnerabilities-in-graphql-role-assignment-ct-editing-and-drafts-tooltips),
unless you can verify based on your log files that the vulnerability has not been exploited,
you should [revoke passwords](https://doc.ibexa.co/en/latest/users/user_management/#revoking-passwords) for all affected users.

## Finish update

Finish the update process:

``` bash
composer run post-install-cmd
```

## Run data migration

### Customer Portal self-registration

> If you are using [[= product_name_exp =]] or [[= product_name_com =]],

Reviewer: ```suggestion
If you are using Ibexa Experience or Ibexa Commerce,
```
It's better not to use those variables in update, because they get messed up in earlier branches.

---
description: Update procedure to v4.4 for people who don't use Commerce packages and can remove them.
---
# Update with new Commerce packages

This update procedure applies if you have a v4.3 installation, and you do not use Commerce packages.

## Update from v4.3.x to v4.3.latest

Before you update to v4.4, you need to go through the following steps to update to the latest maintenance release of v4.3 (v[[= latest_tag_4_3 =]]).

### Update the application to v4.3.latest

Run:

=== "[[= product_name_content =]]"

    ``` bash
    composer require ibexa/content:[[= latest_tag_4_3 =]] --with-all-dependencies --no-scripts
    ```
=== "[[= product_name_exp =]]"

    ``` bash
    composer require ibexa/experience:[[= latest_tag_4_3 =]] --with-all-dependencies --no-scripts
    ```
=== "[[= product_name_com =]]"

    ``` bash
    composer require ibexa/commerce:[[= latest_tag_4_3 =]] --with-all-dependencies --no-scripts
    ```

## Remove deprecated Field Types

By default, every v4.3 installation has a set of built-in Content Types.

> Some of them use Field Types deprecated in v4.4, and they need to be removed manually.

Reviewer: ```suggestion
Some of them use Field Types deprecated in v4.4, which need to be removed manually.
```
otherwise it sounds like removing the Content Types, for Fields

Make sure to remove all occurrences of `sesspecificationstype`, `uivarvarianttype`, `sesselection`, `sesprofiledata` Field Types from your Content Types.

> This step should be performed on working installation, passing it over will result in an error during update:

Reviewer: ```suggestion
This step should be performed on the working installation, omitting it will result in an error during update:
```
